### PR TITLE
Adding hard-coded restrictions to generics (clean history)

### PR DESCRIPTION
**Pull request by ansharlubis**

### CODE REVIEW

File: src/libasr/pass/instantiate_template.cpp [hunk 1]
```diff
@@ -191,28 +227,26 @@ class FunctionInstantiator : public ASR::BaseExprStmtDuplicator<FunctionInstanti
                     ASR::Real_t* tnew = ASR::down_cast<ASR::Real_t>(t);
                     return ASRUtils::TYPE(ASR::make_Real_t(al, t->base.loc,
                             tnew->m_kind, param->m_dims, param->n_dims));
-                }  
+                }
```
Comment (certik) [inline src/libasr/pass/instantiate_template.cpp:230]: You should configure your editor to remove white space, then there is no issue. I am going to merge it, and you can fix it later.

**Review by certik — APPROVED** [review]
This looks good. I am going to merge it, and we can iterate upon it with subsequent PRs.

Thank you so much @ansharlubis for getting this done. This is very very good foundation.

**Comment by certik** [comment]
The next step is to implement something like this:

* https://github.com/lcompilers/lpython/issues/984

and also do all the restrictions checking in `libasr`, that way we can share it with LFortran.

**Comment by ansharlubis** [comment]
Got it!

**Pull request merged**